include/distributed_graph_generator.h

```diff
@@ -0,0 +1,152 @@
+#pragma once
+
+#include <bitset>
+#include <unordered_map>
+#include <variant>
+
+#include "command_graph.h"
+#include "ranges.h"
+#include "region_map.h"
+#include "types.h"
+
+namespace celerity::detail {
+
+class task;
+class task_manager;
+class abstract_command;
+
+// TODO: Make compile-time configurable
+constexpr size_t max_num_nodes = 256;
+using node_bitset = std::bitset<max_num_nodes>;
+
+/**
+ * write_command_state is a command_id with two bits of additional information:
+ *   - Whether the data written by this command is globally still the newest version ("fresh" or "stale")
+ *   - Whether this data has been replicated from somewhere else, i.e., we are not the original producer
+ */
+class write_command_state {
+	constexpr static int64_t fresh_bit = 1ll << 63;
+	constexpr static int64_t replicated_bit = 1ll << 62;
+	static_assert(sizeof(fresh_bit) == sizeof(command_id));
+
+  public:
+	constexpr write_command_state() = default;
```

Comment:
Should this zero-initialize `m_cid` so that at least none of the flags are accidentally set?

Reply:
Yes it should!